src/digraph.jl

```diff
@@ -98,7 +104,15 @@ function add_vertex!(g::DiGraph)
 end
 
 
-has_edge(g::DiGraph, e::Edge) = e in edges(g)
+function has_edge(g::DiGraph, e::Edge)
+    u, v = e
+    u > nv(g) || v > nv(g) && return false
+    if degree(g,u) < degree(g,v)
+        return v in fadj(g,u)
```

Comment:
It's not a bug. It will return the first value greater than or equal to x, but that doesn't mean that x has to exist in the vector in the first place.